pkg/vpcmodel/semanticDiffSubnets.go

```diff
@@ -213,10 +188,325 @@ func diffDescription(diff DiffType) string {
+		if err != nil {
+			return nil, err
+		}
+		disjointContained := false
+		for _, disjointIPBlock := range disjointIPblocks {
+			if disjointIPBlock.ContainedIn(nodeIPBlock) {
+				disjointContained = true
+				for _, thisCidr := range disjointIPBlock.ToCidrList() {
+					newNode := NewExternalNodeForCidr(thisCidr)
+					newNodes = append(newNodes, newNode)
+				}
+			}
+		}
+		if !disjointContained {
+			newNodes = append(newNodes, oldNode)
+		}
+	}
+	return newNodes, nil
+}
+
+func (subnetConnectivity *SubnetConnectivityMap) actualAlignSrcOrDstGivenIPBlists(config *VPCConfig,
+	disjointIPblocks []*common.IPBlock, resizeSrc bool) (
+	alignedConnectivity SubnetConnectivityMap, err error) {
+	// goes over all sources of connections in connectivity
+	// if src is external then for each IPBlock in disjointIPblocks copies dsts and connection type
+	// otherwise just copies as is
+	err = nil
+	alignedConnectivity = map[EndpointElem]map[EndpointElem]*common.ConnectionSet{}
+	for src, endpointConns := range *subnetConnectivity {
+		for dst, conns := range endpointConns {
+			if conns.IsEmpty() {
+				continue
+			}
+			if _, ok := alignedConnectivity[src]; !ok {
+				alignedConnectivity[src] = map[EndpointElem]*common.ConnectionSet{}
+			}
+			// the resizing element is not external - copy as is
+			if (resizeSrc && !src.IsExternal()) || (!resizeSrc && !dst.IsExternal()) {
+				alignedConnectivity[src][dst] = conns
+				continue
+			}
+			// the resizing element is external - go over all ipBlock and allocates the connection
+			// if the ipBlock is contained in the original src/dst
+			var origIPBlock *common.IPBlock
+			if resizeSrc {
+				origIPBlock, err = externalNodeToIPBlock(src.(Node))
+			} else {
+				origIPBlock, err = externalNodeToIPBlock(dst.(Node))
+			}
+			if err != nil {
+				return nil, err
+			}
+			err = config.addIPBlockToConnectivityMap(disjointIPblocks, origIPBlock, alignedConnectivity, src, dst, conns, resizeSrc)
+		}
+	}
+	return alignedConnectivity, err
+}
+
+func (c *VPCConfig) addIPBlockToConnectivityMap(disjointIPblocks []*common.IPBlock,
+	origIPBlock *common.IPBlock, alignedConnectivity map[EndpointElem]map[EndpointElem]*common.ConnectionSet,
+	src, dst EndpointElem, conns *common.ConnectionSet, resizeSrc bool) error {
+	origBlockContainsIPBlocks := false
+	for _, ipBlock := range disjointIPblocks {
+		// 1. get ipBlock of resized index (src/dst)
+		if !ipBlock.ContainedIn(origIPBlock) { // ipBlock not relevant here
+			continue
+		}
+		origBlockContainsIPBlocks = true
+		cidrList := ipBlock.ToCidrList()
+		for _, cidr := range cidrList {
+			nodeOfCidr, err := findNodeWithCidr(c.Nodes, cidr)
+			if err != nil {
+				return err
+			}
+			if resizeSrc {
+				if _, ok := alignedConnectivity[nodeOfCidr]; !ok {
+					alignedConnectivity[nodeOfCidr] = map[EndpointElem]*common.ConnectionSet{}
+				}
+				alignedConnectivity[nodeOfCidr][dst] = conns
+			} else {
+				alignedConnectivity[src][nodeOfCidr] = conns
```

Comment:
It was not needed before because `src` was added to `alignedConnectivity` anyways; but you commented on that and I fixed and indeed now it is checked and added here if needed